cmds/core/mount/mount_unix.go

```diff
@@ -0,0 +1,182 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build freebsd
+
+// mount mounts a filesystem at the specified path.
+//
+// Synopsis:
+//
+//	mount [-r] [-o options] [-t FSTYPE] DEV PATH
+//
+// Options:
+//
+//	-r: read only
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"slices"
+	"strings"
+	"syscall"
+	"unsafe"
+
+	"github.com/u-root/u-root/pkg/mount"
+	"golang.org/x/sys/unix"
+)
+
+var (
+	opts           = map[string]uintptr{}
+	errUsage       = errors.New("")
+	errMountPath   = errors.New("can not read mount path")
+	mountflagsbits = map[uint64]string{
+		unix.MNT_RDONLY:      "Rdonly",
+		unix.MNT_NOEXEC:      "Noexec",
+		unix.MNT_NOSUID:      "Nosuid",
+		unix.MNT_NOATIME:     "Noatime",
+		unix.MNT_SNAPSHOT:    "Snapshot",
+		unix.MNT_SUIDDIR:     "Suiddir",
+		unix.MNT_SYNCHRONOUS: "Synchronous",
+		unix.MNT_ASYNC:       "Async",
+		unix.MNT_FORCE:       "Force",
+		unix.MNT_NOCLUSTERR:  "Noclusterr",
+		unix.MNT_NOCLUSTERW:  "Noclusterw",
+		// unix.MNT_NOCOVER: "Nocover",
+		// unix.MNT_EMPTYDIR: "Emptydir",
+	}
+)
+
+type mountOptions []string
+
+func mountflags(flags uint64) (s string) {
+	if flags == 0 {
+		return ""
+	}
+	for k, v := range mountflagsbits {
+		if (flags & k) == 0 {
+			continue
+		}
+		s += "," + v
+		flags &= ^k
+	}
+	if flags != 0 {
+		s += fmt.Sprintf(",%#x", flags)
+	}
+
+	return
+}
+
+func i8tostring(i []int8) string {
+	x := slices.Index(i, 0)
+	b := *(*[]byte)(unsafe.Pointer(&i))
+	return string(b[:x])
+}
+
+func (o *mountOptions) String() string {
+	return strings.Join(*o, ",")
+}
+
+func (o *mountOptions) Set(value string) error {
```

Comment:
If mountOptions is not trying to implement interface I would remove error, because it always nil.

Reply:
it has to implement the interface however.